### PR TITLE
Event filtering release notes

### PR DESCRIPTION
**Pull request by peterfpeterson**
This adds overall release notes related to event filtering.

**To test:**

`ninja docs-html` and look at the release notes

Refs #34794

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.7.0/framework.rst [hunk 1]
```diff
@@ -5,6 +5,22 @@ Framework Changes
...
+The main changes to mantid are as followed:
+
+1. A ``TimeROI`` object that contains a list of use and ignore regions in time. The times are ``[inclusive, exclusive)``
+2. When filtering/splitting logs in the ``Run`` object, the logs are copied as is and a new ``TimeROI`` object is set on the ``Run``. The main change is that logs no longer have fake values to mimic the filtering
+3. Code should change from asking a ``TimeSeriesProperty`` for its statistics to asking the ``Run`` object for the statistics of a log because the ``Run`` is the only place that knows the information about the ``TimeROI``
+4. When filtering/splitting logs, the filters are now ``[inclusive, exclusive)`` where previous behavior was ``[inclusive, inclusive]``. The previous behavior was inconsistent with how events are filtered. This change, fixes issues observed with the integrated proton charge
+5. When workspaces are added, the resulting ``TimeROI`` is the union of the individual ``TimeROI``. If either workspace had an empty ``TimeROI`` it is assumed to be from that workspace's start-time to end-time.
```
Comment (bolotovskyr) [inline docs/source/release/v6.7.0/framework.rst:18]: @jmborr , is the word "empty" wrt ``TimeROI`` used correctly in this context?

Comment (jmborr) [reply docs/source/release/v6.7.0/framework.rst:18]: @bolotovskyr I was talking to Pete about a very similar issue. I think the wording should be "If either workspace had a  ``TimeROI``  initialized as ``USE_ALL``, it is assumed to be from that workspace's start-time to end-time".

Comment (peterfpeterson) [reply docs/source/release/v6.7.0/framework.rst:18]: We ended up changing that method name in #35424. I will update the PR shortly.

File: docs/source/release/v6.7.0/framework.rst [hunk 2]
```diff
@@ -5,6 +5,22 @@ Framework Changes
 .. contents:: Table of Contents
    :local:
 
+Event Filtering
+---------------
+
+During the v6.7 development cycle, there were major changes to the code underlying event filtering that was originally designed `here <https://github.com/mantidproject/mantid/issues/34794>`_.
+The main changes to mantid are as followed:
```
Comment (bolotovskyr) [inline docs/source/release/v6.7.0/framework.rst:12]: "as follows"?

File: docs/source/release/v6.7.0/framework.rst [hunk 3]
```diff
@@ -5,6 +5,22 @@ Framework Changes
...
+
+1. A ``TimeROI`` object that contains a list of use and ignore regions in time. The times are ``[inclusive, exclusive)``
+2. When filtering/splitting logs in the ``Run`` object, the logs are copied as is and a new ``TimeROI`` object is set on the ``Run``. The main change is that logs no longer have fake values to mimic the filtering
+3. Code should change from asking a ``TimeSeriesProperty`` for its statistics to asking the ``Run`` object for the statistics of a log because the ``Run`` is the only place that knows the information about the ``TimeROI``
+4. When filtering/splitting logs, the filters are now ``[inclusive, exclusive)`` where previous behavior was ``[inclusive, inclusive]``. The previous behavior was inconsistent with how events are filtered. This change, fixes issues observed with the integrated proton charge
+5. When workspaces are added, the resulting ``TimeROI`` is the union of the individual ``TimeROI``. If either workspace had a ``TimeROI`` with ``TimeROI.useAll()==True``, it is assumed to be from that workspace's start-time to end-time.
```
Comment (bolotovskyr) [inline docs/source/release/v6.7.0/framework.rst:18]: "of the individual ``TimeROI``s" ?

File: docs/source/release/v6.7.0/framework.rst [hunk 4]
```diff
@@ -5,6 +5,22 @@ Framework Changes
...
 
+Event Filtering
+---------------
+
+During the v6.7 development cycle, there were major changes to the code underlying event filtering that was originally designed `here <https://github.com/mantidproject/mantid/issues/34794>`_.
+The main changes to mantid are as followed:
+
+1. A ``TimeROI`` object that contains a list of use and ignore regions in time. The times are ``[inclusive, exclusive)``
+2. When filtering/splitting logs in the ``Run`` object, the logs are copied as is and a new ``TimeROI`` object is set on the ``Run``. The main change is that logs no longer have fake values to mimic the filtering
+3. Code should change from asking a ``TimeSeriesProperty`` for its statistics to asking the ``Run`` object for the statistics of a log because the ``Run`` is the only place that knows the information about the ``TimeROI``
```
Comment (bolotovskyr) [inline docs/source/release/v6.7.0/framework.rst:16]: Missing period at the end of the sentence.

File: docs/source/release/v6.7.0/framework.rst [hunk 5]
```diff
@@ -5,6 +5,22 @@ Framework Changes
...
+4. When filtering/splitting logs, the filters are now ``[inclusive, exclusive)`` where previous behavior was ``[inclusive, inclusive]``. The previous behavior was inconsistent with how events are filtered. This change, fixes issues observed with the integrated proton charge
```
Comment (bolotovskyr) [inline docs/source/release/v6.7.0/framework.rst:17]: Missing period at the end of the sentence.

**Review by bolotovskyr — APPROVED** [review]
A good summary of the new functionality.

**Pull request merged**